static/app/views/monitors/components/checkInTimeline.tsx

```diff
@@ -0,0 +1,63 @@
+import styled from '@emotion/styled';
+
+import {Resizeable} from 'sentry/components/replays/resizeable';
+import {space} from 'sentry/styles/space';
+import {CheckIn, CheckInStatus} from 'sentry/views/monitors/types';
+
+interface Props {
+  checkins: CheckIn[];
+  end: Date;
+  start: Date;
+}
+
+function getCheckInPosition(checkDate: string, timelineStart: Date, msPerPixel: number) {
+  const elapsedSinceStart = new Date(checkDate).getTime() - timelineStart.getTime();
+  return elapsedSinceStart / msPerPixel;
+}
+
+export function CheckInTimeline({checkins, start, end}: Props) {
+  return (
+    <Resizeable>
+      {({width}) => {
+        const timeWindow = end.getTime() - start.getTime();
+        const msPerPixel = timeWindow / width;
+
+        return (
+          <TimelineContainer>
+            {checkins.map(({id, dateCreated, status}) => {
+              const left = getCheckInPosition(dateCreated, start, msPerPixel);
+              if (left < 0) {
+                return null;
+              }
+
+              return <JobTick key={id} left={left} status={status} />;
+            })}
+          </TimelineContainer>
+        );
+      }}
+    </Resizeable>
+  );
+}
+
+const TimelineContainer = styled('div')`
+  position: relative;
+  height: 14px;
+  margin: ${space(4)} 0;
+`;
+
+const JobTick = styled('div')<{left: number; status: CheckInStatus}>`
+  position: absolute;
+  width: 4px;
+  height: 14px;
+  border-radius: 6px;
+  left: ${p => p.left}px;
+
+  background: ${p =>
+    p.status === CheckInStatus.ERROR || p.status === CheckInStatus.TIMEOUT
+      ? p.theme.red200
+      : p.status === CheckInStatus.OK
+      ? p.theme.green200
+      : p.status === CheckInStatus.MISSED
+      ? p.theme.yellow200
+      : p.theme.disabled};
```

Comment:
Instead I would have a mapping of `Record<CheckInStatus, Color>` and read from that